### DISCUSSION TITLE
How to get type safe return type instead of empty object {} when passing select as variable?

### DISCUSSION THREAD

**Discussion by DoronTorangy** [prisma/prisma#19120] [Q&A]
I have the following function:
```ts
async getUserById(id: string, select?: Prisma.UserSelect):  {
	return this.prismaService.user.findUniqueOrThrow({ where: { id }, select })
}
```

however when I call that function, typescript cannot infer my select clause, and says the function return type is `{}`.

```ts
const user = await this.getUserById(userId, { firstName: true }) // typescript infers user as {}

user.firstName // Property firstName does not exist on type {}.
```

I tried using `Prisma.validator` and also following [this issue](https://github.com/prisma/prisma/issues/3372), but nothing works.

anyone knows how to fix that?

**Comment by nurul3101** [comment]
Hello @DoronTorangy 👋 

Can you share how did you use the Prisma.validator method which didn't fix the issue?

**Reply by nurul3101** [reply]
Please let us know in case you still need any help.

**Reply by nurul3101** [reply]
I'll mark this discussion as closed, please feel free to reopen in case you need any assistance.

**Comment by charliematters** [comment]
I'm seeing this behaviour in a remix project that I'm migrating from v1 to v2. I'm not sure which change has caused it, but any time I use a `select` argument I'm getting `{}` as opposed to the correct types. Causing quite a headache!

I've rm-rf-ed my node_modules, I've rebuilt the Prisma client and I've updated everything to the latest version (prisma 5.4.2, typescript 5.2.2), but it's still not happy. I'll work on cutting it down, but it's a really simple case.

![image](https://github.com/prisma/prisma/assets/7063124/a4f20bb4-b436-4de2-905f-447e2dab09c5)
